helix-term/src/commands.rs

```diff
@@ -1000,31 +1000,70 @@ fn goto_file_vsplit(cx: &mut Context) {
     goto_file_impl(cx, Action::VerticalSplit);
 }
 
+fn resolve_selected_path(
+    selection_text: String,
+    doc_path_parent: Option<&Path>,
+) -> Option<PathBuf> {
+    let trimmed = selection_text.trim();
+    if trimmed.is_empty() {
+        return None;
+    }
+
+    let selected_path = PathBuf::from(trimmed);
+    match (selected_path.is_relative(), doc_path_parent) {
+        (true, Some(parent_path)) => {
+            let mut result = PathBuf::new();
+            result.push(parent_path);
+            result.push(selected_path);
+            Some(result)
```

Comment:
```suggestion
            Some(parent_path + selected_path)
```
Would this work?

Reply:
I can test it, but I would expect `PathBuf::push` to handle directory seperators correctly in all cases and I am afraid, `parent_path + selected_path` would not.

Reply:
How about `Some(parent_path.join(selected_path))`?

Reply:
Looks good. Will do, when I am back from work.